resolver/serviceResolver.go

```diff
@@ -432,6 +419,60 @@ func (resolver *serviceResolver) validateTxRequestReturningGuardian(
 	return guardianInfo, otpVerifyCodeData, nil
 }
 
+func (resolver *serviceResolver) checkAllowanceAndVerifyCode(
+	userInfo *core.UserInfo,
+	userAddress string,
+	userIp string,
+	code string,
+	secondCode string,
+	guardianAddr []byte,
+) (*requests.OTPCodeVerifyData, error) {
+	verifyCodeData, err := resolver.secureOtpHandler.IsVerificationAllowedAndIncreaseTrials(userAddress, userIp)
+	if err != nil {
+		return verifyCodeData, err
+	}
+
+	err = resolver.verifyCode(userInfo, code, guardianAddr)
+	if err != nil {
+		return verifyCodeData, err
+	}
+	resolver.secureOtpHandler.Reset(userAddress, userIp)
+
+	err = resolver.verifySecurityModeCode(userInfo, userAddress, secondCode, guardianAddr, verifyCodeData.SecurityModeRemainingTrials)
+	remainingSecurityTrials := verifyCodeData.SecurityModeRemainingTrials
+	if err != nil {
+		remainingSecurityTrials--
+	}
+	return &requests.OTPCodeVerifyData{
+		RemainingTrials:             int(resolver.secureOtpHandler.FreezeMaxFailures()),
+		ResetAfter:                  0,
+		SecurityModeRemainingTrials: remainingSecurityTrials, // decrementing failed trials increases remaining trials
```

Comment:
i think `SecurityModeRemainingTrials` might end up being below 0 after decrementing on error if `verifyCodeData.SecurityModeRemainingTrials` will be 0; maybe a check for `remainingSecurityTrials` to be decremented only if > 0

Reply:
done